src/apps/views/AppDetails/AppDetails.tsx

```diff
@@ -1,23 +1,94 @@
-import Container from "@saleor/components/Container";
-import PageHeader from "@saleor/components/PageHeader";
-import { sectionNames } from "@saleor/intl";
-import React from "react";
-import { useIntl } from "react-intl";
+import useNavigator from "@saleor/hooks/useNavigator";
+import createDialogActionHandlers from "@saleor/utils/handlers/dialogActionHandlers";
+import React, { useEffect } from "react";
 
-import { AppListUrlQueryParams } from "../../urls";
+import AppActivateDialog from "../../components/AppActivateDialog";
+import AppDeactivateDialog from "../../components/AppDeactivateDialog";
+import AppDetailsPage from "../../components/AppDetailsPage";
+import {
+  useAppActivateMutation,
+  useAppDeactivateMutation
+} from "../../mutations";
+import { useAppDetails } from "../../queries";
+import {
+  AppDetailsUrlDialog,
+  AppDetailsUrlQueryParams,
+  appsListPath,
+  appUrl
+} from "../../urls";
 
 interface AppDetailsProps {
   id: string;
-  params: AppListUrlQueryParams;
+  params: AppDetailsUrlQueryParams;
 }
 
-export const AppDetails: React.FC<AppDetailsProps> = () => {
-  const intl = useIntl();
+export const AppDetails: React.FC<AppDetailsProps> = ({ id, params }) => {
+  const { data, loading, refetch } = useAppDetails({
+    displayLoader: true,
+    variables: { id }
+  });
+  const navigate = useNavigator();
+  const mutationOpts = { variables: { id } };
+  const [activateApp, activateAppResult] = useAppActivateMutation({});
+  const [deactivateApp, deactivateAppResult] = useAppDeactivateMutation({});
+
+  const [openActivateModal, closeActivateModal] = createDialogActionHandlers<
+    AppDetailsUrlDialog,
+    AppDetailsUrlQueryParams
+  >(navigate, params => appUrl(id, params), params);
+
+  const [
+    openDeactivateModal,
+    closeDeactivateModal
+  ] = createDialogActionHandlers<AppDetailsUrlDialog, AppDetailsUrlQueryParams>(
+    navigate,
+    params => appUrl(id, params),
+    params
+  );
+
+  useEffect(() => {
+    const { status, data } = activateAppResult;
+    if (status === "success" && data?.appActivate?.errors.length === 0) {
+      refetch();
+      closeActivateModal();
+    }
+  }, [activateAppResult.status]);
```

Comment:
Mutations accept object containing success and error handlers as an argument, so no need for `useEffect` here if we can do it more declarative way